.circleci/config.yml

```diff
@@ -101,7 +109,23 @@ jobs:
           name: Testing
           command: make trace_test
           no_output_timeout: 1800s
-
+      - restore_cache:
+          name: Restoring ./target Cache
```

Comment:
Can this be merged with L83?

Reply:
Unlike kcov, which can directly work on executables. Tarpaulin needs to rebuild the project to instrument breakpoints into executables so that it can trace their execution. Therefore, we need to save the cache after running tarpaulin.

The current workflow is: restore cache -> build -> save cache -> test -> restore cache -> coverage -> save cache
After merge: restore cache -> build -> test -> coverage -> save cache

Currently, some test fails randomly, which will stop the job immediately. I am worrying about that in these cases, the built target can not be saved.

Reply:
It's OK. After merge it should be restore cache -> build -> coverage -> save cache. From the timeline, it can reduce at least 40 minutes. I think the cost of rebuild is far less than that.

Reply:
restore cache -> build -> coverage -> save cache -> test ?

Reply:
The test is run when calculating coverage, why run it again?

Reply:
Tarpaulin does run the tests, but it always returns 0 regardless of any test fails or not. Therefore, it can not be used as a test.

Reply:
OK, sounds like a test before calculating coverage is required. But merging still costs less.